examples/Darcy/darcy.jl

```diff
@@ -0,0 +1,179 @@
+
+# We will solve for an unknown permeability field ``\kappa`` governing the velocity of a Darcy flow on a square 2D domain. To learn about the permeability we shall take few pointwise measurements of the solved velocity field within the domain. The forward solver is a simple finite difference scheme taken and modified from code [here](https://github.com/Zhengyu-Huang/InverseProblems.jl/blob/master/Fluid/Darcy-2D.jl). 
+
+# First we load standard packages
+using LinearAlgebra
+using Distributions
+using Random
+using JLD2
+
+# the package to define the function distributions
+import GaussianRandomFields # we wrap this so we don't want to use "using"
+const GRF = GaussianRandomFields
+
+# and finally the EKP packages
+using EnsembleKalmanProcesses
+using EnsembleKalmanProcesses.ParameterDistributions
+const EKP = EnsembleKalmanProcesses
+
+# We include the forward solver here
+include("GModel.jl")
+
+# Then link some outputs for figures and plotting
+fig_save_directory = joinpath(@__DIR__, "output")
+data_save_directory = joinpath(@__DIR__, "output")
+if !isdir(fig_save_directory)
+    mkdir(fig_save_directory)
+end
+if !isdir(data_save_directory)
+    mkdir(data_save_directory)
+end# TOML interface for fitting parameters of a sinusoid
+
+PLOT_FLAG = true
+if PLOT_FLAG
+    using Plots
+    @info "Plotting enabled, this will reduce code performance. Figures stored in $fig_save_directory"
+end
+
+# Set a random seed.
+seed = 100234
+rng = Random.MersenneTwister(seed)
+
+# Define the spatial domain and discretization 
+dim = 2
+N, L = 80, 1.0
+pts_per_dim = LinRange(0, L, N)
+obs_ΔN = 10
+
+# To provide a simple test case, we assume that the true function parameter is a particular sample from the function space we set up to define our prior. More precisely we choose a value of the truth that doesnt have a vanishingly small probability under the prior defined by a probability distribution over functions; here taken as a family of Gaussian Random Fields (GRF). The function distribution is characterized by a covariance function - here a Matern kernel which assumes a level of smoothness over the samples from the distribution. We define an appropriate expansion of this distribution, here based on the Karhunen-Loeve expansion (similar to an eigenvalue-eigenfunction expansion) that is truncated to a finite number of terms, known as the degrees of freedom (`dofs`). The `dofs` define the effective dimension of the learning problem, decoupled from the spatial discretization. Explicitly, larger `dofs` may be required to represent multiscale functions, but come at an increased dimension of the parameter space and therefore a typical increase in cost and difficulty of the learning problem.
```

Comment:
I think it would be useful to link the GaussianRandomField documentation: https://pieterjanrobbe.github.io/GaussianRandomFields.jl/stable/